### PR TITLE
[Hactoberfest] Fail the execution at the end if any test failures

### PR DESCRIPTION
**Pull request by v1v**
Closes https://github.com/jenkinsci/docker/issues/1009

### What

Fail the build if any test failures happened when running the windows tests.

It does not fail fast to be able to provide as many details as possible.

**NOTE**: Please squash and merge

### Why

It does not fail if test failures at the moment

### Tests

https://github.com/jenkinsci/docker/pull/1011/commits/779f5d9a084c72ca1ef700f3737232240afbd94d finally worked

image

### Issues

Failed tests related to https://github.com/jenkinsci/docker/issues/1014

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]
Seems very reasonable to me.  @slide is the expert on this area, along with possible help from @garethjevans

File: make.ps1 [hunk 1]
```diff
@@ -106,14 +108,25 @@ if($target -eq "test") {
     if(![System.String]::IsNullOrWhiteSpace($Build) -and $builds.ContainsKey($Build)) {
         $env:FOLDER = $builds[$Build]['Folder']
         Invoke-Pester -Path tests -EnableExit
+        if($lastExitCode -ne 0) {
+            testFailed = 1
```
Comment (slide) [inline make.ps1:112]: Missing $?

File: make.ps1 [hunk 2]
```diff
@@ -106,14 +108,25 @@ if($target -eq "test") {
...
+        if($lastExitCode -ne 0) {
+            testFailed = 1
+        }
         Remove-Item -Force env:\FOLDER
     } else {
         foreach($b in $builds.Keys) {
             $env:FOLDER = $builds[$b]['Folder']
             Invoke-Pester -Path tests -EnableExit
+            if($lastExitCode -ne 0) {
+                testFailed = 1
```
Comment (slide) [inline make.ps1:120]: Missing $?

**Review by slide — CHANGES_REQUESTED** [review]

**Comment by v1v** [comment]
I'm testing whether https://github.com/jenkinsci/docker/pull/1011/commits/3d0820fa40c11b58290daf4eeaf0885a40b6e5ae will fail as expected

**Comment by v1v** [comment]
Unfortunately I don't own any windows computer, so I'm testing it whether this change works as expected,  it seems it works in my mind, sorry about this iterations

File: make.ps1 [hunk 3]
```diff
@@ -106,14 +108,28 @@ if($target -eq "test") {
...
+            $testFailed = 1
+        }
         Remove-Item -Force env:\FOLDER
     } else {
         foreach($b in $builds.Keys) {
             $env:FOLDER = $builds[$b]['Folder']
             Invoke-Pester -Path tests -EnableExit
+            if($lastExitCode -ne 0) {
+                Write-Host "Testing $b failed"
+                $testFailed = 1
```
Comment (slide) [inline make.ps1:122]: You could also use $true and $false instead of 1 and 0, but that is not required.

File: make.ps1 [hunk 4]
```diff
@@ -107,26 +107,26 @@ if($target -eq "test") {
...
+            $testFailed = $true
         }
         Remove-Item -Force env:\FOLDER
     } else {
         foreach($b in $builds.Keys) {
             $env:FOLDER = $builds[$b]['Folder']
-            Invoke-Pester -Path tests -EnableExit
-            if($lastExitCode -ne 0) {
+            $r = Invoke-Pester -Path tests -EnableExit
+            if ("Failed" -eq $r.Result) {
```
Comment (v1v) [inline make.ps1:120]: See https://github.com/pester/Pester/blob/3c1b60138c1cde3a51c8db333de41a3920422fe1/test.ps1/#L139-L142

**Comment by v1v** [comment]
I was forced to push another commit since the GH UI merge conflicts changed the EOL:
- https://github.com/jenkinsci/docker/commit/2e76727f466cb03075a033b91f730bd242698e89#diff-8876131e1b9638758221a09dbc928ac5e989b22a4cad16f925cf0b5f27477d53

**Review by slide — APPROVED** [review]

**Pull request merged**